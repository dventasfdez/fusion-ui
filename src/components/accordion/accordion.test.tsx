import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import Accordion, { AccordionContent, AccordionHeader, IAccordionProps } from "./accordion";

const accordionExample = (props?: IAccordionProps) => (
  <Accordion {...props} className="tag-ds" data-testid="accordion">
    <AccordionHeader data-testid="accordion-header">
      Accordion example
      <span className="accordion-helper-text">Helper</span>
    </AccordionHeader>
    <AccordionContent data-testid="accordion-content">
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Dignissimos eaque beatae pariatur maiores excepturi, sequi natus obcaecati vel neque reiciendis, cumque quis alias exercitationem quo
      tempore corrupti odit atque quae, facilis aut molestiae incidunt rerum dolore. Blanditiis quis at est, numquam ad temporibus error iure quae? Eius unde quam accusantium.
    </AccordionContent>
  </Accordion>
);
const accordionExampleWithoutClassName = (props?: IAccordionProps) => (
  <Accordion {...props}>
    <AccordionHeader>
      Accordion example
      <span className="accordion-helper-text">Helper</span>
    </AccordionHeader>
    <AccordionContent>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Dignissimos eaque beatae pariatur maiores excepturi, sequi natus obcaecati vel neque reiciendis, cumque quis alias exercitationem quo
      tempore corrupti odit atque quae, facilis aut molestiae incidunt rerum dolore. Blanditiis quis at est, numquam ad temporibus error iure quae? Eius unde quam accusantium.
    </AccordionContent>
  </Accordion>
);
const accordionExampleWithoutHeader = (props?: IAccordionProps) => (
  <Accordion {...props} className="tag-ds">
    <AccordionContent>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Dignissimos eaque beatae pariatur maiores excepturi, sequi natus obcaecati vel neque reiciendis, cumque quis alias exercitationem quo
      tempore corrupti odit atque quae, facilis aut molestiae incidunt rerum dolore. Blanditiis quis at est, numquam ad temporibus error iure quae? Eius unde quam accusantium.
    </AccordionContent>
  </Accordion>
);

describe("Accordion snapshots", () => {
  it("Accordion should render", () => {
    const { container } = render(accordionExample());

    expect(container).toMatchSnapshot();
  });
});
// test();

// test('Accordion default show content should render', () => {
//   const component = renderer.create(accordionExample({defaultShow: true}));
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('Accordion filled should render', () => {
//   const component = renderer.create(accordionExample({filled: true}));
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('Accordion filled show content should render', () => {
//   const component = renderer.create(accordionExample({defaultShow: true, filled: true}));
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('Accordion without header should render', () => {
//   const component = renderer.create(accordionExampleWithoutHeader());
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('Accordion without class name should render', () => {
//   const component = renderer.create(accordionExampleWithoutClassName());
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('Click accordion header', () => {
//   const {container, getByTestId} = render(accordionExample({'data-testid': 'accordion'}));
//   expect(container.getElementsByClassName('accordion-content').length).toBe(0);
//   const accordionHeader = getByTestId('accordion-header');
//   if (accordionHeader) fireEvent.click(accordionHeader);
//   expect(container.getElementsByClassName('accordion-content').length).toBe(1);
// });

// test('Rerender accordion', () => {
//   const {container, rerender} = render(accordionExample({'data-testid': 'accordion', defaultShow: true}));
//   expect(container.getElementsByClassName('accordion-content').length).toBe(1);
//   rerender(accordionExample({'data-testid': 'accordion', defaultShow: false}));
//   expect(container.getElementsByClassName('accordion-content').length).toBe(0);
// });
