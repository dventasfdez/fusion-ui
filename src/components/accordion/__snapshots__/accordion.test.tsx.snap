// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion default show content should render 1`] = `
<div
  className="accordion tag-ds show "
  data-testid="accordion"
>
  <div
    className="accordion-header "
    data-testid="accordion-header"
    onClick={[Function]}
  >
    Accordion example
    <span
      className="accordion-helper-text"
    >
      Helper
    </span>
    <span
      className="material-icons accordion-icon"
    >
      expand_less
    </span>
  </div>
  <div
    className="accordion-content "
    data-testid="accordion-content"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Dignissimos eaque beatae pariatur maiores excepturi, sequi natus obcaecati vel neque reiciendis, cumque quis alias exercitationem quo tempore corrupti odit atque quae, facilis aut molestiae incidunt rerum dolore. Blanditiis quis at est, numquam ad temporibus error iure quae? Eius unde quam accusantium.
  </div>
</div>
`;

exports[`Accordion filled should render 1`] = `
<div
  className="accordion_filled tag-ds  "
  data-testid="accordion"
>
  <div
    className="accordion-header "
    data-testid="accordion-header"
    onClick={[Function]}
  >
    Accordion example
    <span
      className="accordion-helper-text"
    >
      Helper
    </span>
    <span
      className="material-icons accordion-icon"
    >
      expand_more
    </span>
  </div>
</div>
`;

exports[`Accordion filled show content should render 1`] = `
<div
  className="accordion_filled tag-ds show "
  data-testid="accordion"
>
  <div
    className="accordion-header "
    data-testid="accordion-header"
    onClick={[Function]}
  >
    Accordion example
    <span
      className="accordion-helper-text"
    >
      Helper
    </span>
    <span
      className="material-icons accordion-icon"
    >
      expand_less
    </span>
  </div>
  <div
    className="accordion-content "
    data-testid="accordion-content"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Dignissimos eaque beatae pariatur maiores excepturi, sequi natus obcaecati vel neque reiciendis, cumque quis alias exercitationem quo tempore corrupti odit atque quae, facilis aut molestiae incidunt rerum dolore. Blanditiis quis at est, numquam ad temporibus error iure quae? Eius unde quam accusantium.
  </div>
</div>
`;

exports[`Accordion should render 1`] = `
<div
  className="accordion tag-ds  "
  data-testid="accordion"
>
  <div
    className="accordion-header "
    data-testid="accordion-header"
    onClick={[Function]}
  >
    Accordion example
    <span
      className="accordion-helper-text"
    >
      Helper
    </span>
    <span
      className="material-icons accordion-icon"
    >
      expand_more
    </span>
  </div>
</div>
`;

exports[`Accordion without class name should render 1`] = `
<div
  className="accordion   "
>
  <div
    className="accordion-header "
    onClick={[Function]}
  >
    Accordion example
    <span
      className="accordion-helper-text"
    >
      Helper
    </span>
    <span
      className="material-icons accordion-icon"
    >
      expand_more
    </span>
  </div>
</div>
`;

exports[`Accordion without header should render 1`] = `
<div
  className="accordion tag-ds  "
/>
`;
